WordPress/gutenberg
# [List v2]: Handle Enter in empty list items

## What?
Part of: https://github.com/WordPress/gutenberg/issues/39519
Alternative of: https://github.com/WordPress/gutenberg/pull/39626
Based on @mtias's suggested algorithm here: https://github.com/WordPress/gutenberg/pull/39626#issuecomment-1080555671 

>I think pressing `enter` on an empty list item should backtrack one level at a time. This video is from Apple Notes as an example (I'm only pressing enter):

https://user-images.githubusercontent.com/548849/160392233-d4345663-f3ba-491d-ab7f-1b96456d90e8.mov

In the experimental new list blocks we should split lists on empty list items if we press enter.



## How?
Since my first approach with a different algorithm, the indentation has been implemented by @jorgefilipecosta and I'm using one of the hooks for one of the two parts of the algorithm. 

## Testing Instructions
1. Insert a new list block with any nested lists
2. On an empty list item press `Enter`
3. Should split the list and insert a paragraph in the middle
4. If the list item we press Enter is nested it should outdent

## Screenshots or screencast 

https://user-images.githubusercontent.com/16275880/160637870-3663d9a3-321d-41f7-aced-ab89ee6e0ac6.mov


## Notes
1. The handling of `Enter` in a non empty list item is not handled yet, as it needs more changes involving the RichText API.
2. I've created a new `hooks` file and moved `useIndentListItem and useOutdentListItem` without making any change. The new hook is the `useEnter` one.
